components/nimbus/src/evaluator.rs

```diff
@@ -51,27 +51,28 @@ pub struct TargetingAttributes {
     #[serde(flatten)]
     pub app_context: AppContext,
     #[serde(flatten)]
-    pub request_context: RequestContext,
+    pub request_context: Map<String, Value>,
     pub language: Option<String>,
     pub region: Option<String>,
-    pub os: Option<String>,
 }
 
 #[cfg(not(feature = "stateful"))]
 impl TargetingAttributes {
-    pub fn new(app_context: AppContext, request_context: RequestContext) -> Self {
-        let (language, region) = request_context
-            .locale
-            .clone()
-            .map(split_locale)
-            .unwrap_or_else(|| (None, None));
+    pub fn new(app_context: AppContext, request_context: Map<String, Value>) -> Self {
+        let (language, region) = match request_context
+            .get("locale")
```

Comment:
So `locale` is required in the request context?  Is there any way to remove it as a special case and not put any hard requirements on the structure of the targeting context?

Reply:
Nope it's not required! I left it as optional so that if they choose to include it we parse it like we do for iOS/Android.

Just to quickly walk through the control flow here:
```rust
let (language, region) = match request_context // perform a `match` on the result of this execution
            .get("locale")            // `get` returns an Option and does not error on missing key
            .unwrap_or(&Value::Null)  // `unwrap_or` unwraps if it's there, and uses a different value if it's not. This changes the value in the execution chain from an `Option<Value>` to `Value` — but if it _wasn't_ present the `Value` will be JSON-equiv null
            .as_str()                 // `as_str` returns `Value` instance as an optional Rust `str` type if the `Value` is a string, otherwise it returns the option as `None`
        {
            Some(locale) => split_locale(locale.to_string()), // Here we handle the split if the string is present
            _ => (None, None),          // Here we handle the case where it's not, setting lang/region to `None`
        };
```

Though if we're not a fan of having any amount of extra logic around the targeting for Cirrus I can just remove this and the lang/region members entirely.

Reply:
I think eventually they will be using locale, they use it in unleash too